typescript/cli/src/utils/keys.ts

```diff
@@ -14,34 +12,32 @@ const IMPERSONATED_KEY_TYPE = 'address';
  * Retrieves a signer for the current command-context.
  * @returns the signer
  */
-export async function getSigner<P extends ContextSettings>({
-  keyConfig,
+export async function getSigner({
+  key,
   skipConfirmation,
-}: P): Promise<providers.JsonRpcSigner | ethers.Wallet | undefined> {
-  if (!keyConfig) return undefined;
-
-  const key = await retrieveKey(DEFAULT_KEY_TYPE, keyConfig, skipConfirmation);
-
-  return privateKeyToSigner(key);
+}: {
+  key?: string;
+  skipConfirmation?: boolean;
+}) {
+  key ||= await retrieveKey(DEFAULT_KEY_TYPE, skipConfirmation);
+  const signer = privateKeyToSigner(key);
+  return { key, signer };
 }
 
 /**
  * Retrieves an impersonated signer for the current command-context.
  * @returns the impersonated signer
  */
-export async function getImpersonatedSigner<P extends ContextSettings>({
-  keyConfig,
+export async function getImpersonatedSigner({
+  key,
   skipConfirmation,
-}: P): Promise<providers.JsonRpcSigner | ethers.Wallet | undefined> {
-  if (!keyConfig) return undefined;
-
-  const key = await retrieveKey(
-    IMPERSONATED_KEY_TYPE,
-    keyConfig,
-    skipConfirmation,
-  );
-
-  return await addressToImpersonatedSigner(key);
+}: {
+  key?: string;
+  skipConfirmation?: boolean;
+}) {
+  key ||= await retrieveKey(IMPERSONATED_KEY_TYPE, skipConfirmation);
```

Comment:
impersonated key type should never prompt for private key fwiw

Reply:
It prompts for an address